probatus/missing_values/imputation.py

```diff
@@ -0,0 +1,384 @@
+
+            n_jobs (int, optional):
+                Number of cores to run in parallel while fitting across folds. None means 1 unless in a
+                `joblib.parallel_backend` context. -1 means using all processors.
+
+            verbose (int, optional):
+                Controls verbosity of the output:
+
+                - 0 - nether prints nor warnings are shown
+                - 1 - 50 - only most important warnings regarding data properties are shown (excluding SHAP warnings)
+                - 51 - 100 - shows most important warnings, prints of the feature removal process
+                - above 100 - presents all prints and all warnings (including SHAP warnings).
+
+            random_state (int, optional):
+                Random state set at each round of feature elimination. If it is None, the results will not be
+                reproducible and in random search at each iteration a different hyperparameters might be tested. For
+                reproducible results set it to integer.
+        """
+        self.clf = clf
+        self.model_na_support = model_na_support
+        self.scorer = get_single_scorer(scoring)
+        self.strategies = strategies
+        self.cv = cv
+        self.verbose = verbose
+        self.n_jobs = n_jobs
+        self.random_state = random_state
+        self.fitted = False
+        self.report_df = pd.DataFrame(
+            columns=[
+                "strategy",
+                f"{self.scorer.metric_name} score",
+                f"{self.scorer.metric_name} std",
+            ]
+        )
+
+    def __repr__(self):
+        return "Imputation comparision for {}".format(self.clf.__class__.__name__)
+
+    def fit(self, X, y, column_names=None):
+        """
+        Calculates the cross validated results for various imputation strategies.
+
+        Args:
+            X (pd.DataFrame):
+                input variables.
+
+            y (pd.Series):
+                target variable.
+
+            column_names (None, or list of str, optional):
+                List of feature names for the dataset.
+                If None, then column names from the X dataframe are used.
+        """
+        if self.random_state is not None:
+            np.random.seed(self.random_state)
+
+        # Place holder for results.
+        results = []
+
+        self.X, self.column_names = preprocess_data(
+            X, column_names=column_names, verbose=self.verbose
+        )
+        self.y = preprocess_labels(y, index=self.X.index, verbose=self.verbose)
+
+        # Identify categorical features.
+        X_cat = X.select_dtypes(include=["category", "object"])
+        categorical_columns = X_cat.columns.to_list()
+
+        # Identify the numeric columns.Numeric columns are all columns expect the categorical columns
+        
+        X_num = X.drop(columns=categorical_columns, inplace=False)
+        numeric_columns = X_num.columns.to_list()
```

Comment:
I would suggest using this instead of the two lines:
```
df.select_dtypes('number').columns
```

Reply:
Good idea.Doen.